api/pom.xml

```diff
@@ -83,6 +90,21 @@ limitations under the License.
             <artifactId>logback-classic</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>3.6</version>
+        </dependency>
```

Comment:
Is the Pair stuff in AutomataLib better? Didn't I read somewhere in AutomataLib that it is deprecated?

Reply:
It's better in the sense, that it doesn't pull a new dependency for already existing functionality.

`Triple` was deprecated and removed (although the reasoning for this may also be applicable to the `Pair` class). I'll have to think about this. I understand (and agree), that generic Tuple classes are not really expressive, but your use-cases showed that it is sometimes rather inconvenient to write a custom inner class for every aggregation of values. Maybe we'll state in the JavaDoc that `Pair` (and maybe `Triple` will make a comeback) should only be used for internal data storage and any exposed/returned data aggregation should be wrapped in a seperate (expressive) class.

As for this issue though, it should be a drop-in replacement with the AutomataLib versions.

Reply:
I agree that you should not use it as a return type. And I don't mind which to use, but why (re-)introduce Pair and Triple in AutomataLib while it is already in Apache commons?

Reply:
Well, it's a question of "do we really want to pull a (big) dependency for something trivial?". [This blog post](http://www.haneycodes.net/npm-left-pad-have-we-forgotten-how-to-program/) is actually a funny rant about this topic. Since we already have a `Pair` class, I wouldn't introduce a redundant dependency for this.

Reply:
Okay, for now I can use Pair in automatalib.